setup.py

```diff
@@ -66,7 +66,7 @@ def get_transforms_data():
 }
 
 INSTALL_REQUIRES = [
-    'numpy',
+    'numpy>=1.21',
```

Comment:
Why this restriction?

Reply:
NEP 29 states that only numpy 1.21+ versions are supported.

Reply:
So what? I think we should put restrictions if our code uses some features of `numpy` that _require_ specific `numpy` version. I do not think it is our business to enforce what NEP states. If an user wants to use an unsupported Python or `numpy` version, why we should not allow it?

Reply:
Regardless of what version tweakwcs chooses to pin numpy, there is some oldest version of numpy that it functions with and it should pin its lower dependency to that as that's a core dependency. Moreover, there is an implicit lower pin on numpy versions coming from astropy at the very least. It's much easier for end users to figure out these things if tweakwcs is explicit about its pinning instead of relying on its other dependencies to do this. 

The simplest solution from a support point of view is to lower pin dependencies to what numpy is currently supporting. This way tweakwcs does not have an obligation to attempt fixes due to old versions of numpy.

Reply:
Except for the first sentence, I disagree with most of this statement.

For example, "implicit lower pin ... coming from astropy" - why do I have to make an implicit pin explicit? Shouldn't `pip` take care of this?

"It's much easier for end users to figure out these things if tweakwcs is explicit about ..." Again, I disagree. Most users do not look into dependency versions if `pip` does that for them. If someone wants to test `tweakwcs` with an older version of `numpy` or `astropy` I do not see why I should not allow this.

"... tweakwcs does not have an obligation to attempt fixes due to old versions of numpy". The likelihood of tweakreg suddenly stopping working with an old version is small.

Reply:
I completely disagree with this sentiment, but I am not the maintainer of this package so I'll leave the choice of how to fix this to you.